File: javascript/bob/bob.js
export const hey = (message) => {
  message = message.trim();
  if (message === '') return 'Fine. Be that way!';
  if (/.*[a-z].*/i.test(message) && message.toUpperCase() === message) {
    if (message.endsWith('?')) return 'Calm down, I know what I\'m doing!';
    else return 'Whoa, chill out!';
  }
  if (message.endsWith('?')) return 'Sure.';
  return 'Whatever.';
};

File: kotlin/bob/src/main/kotlin/Bob.kt
package Bob

fun hey(s: String): String {
  var trimmed = s.trim();
  if (trimmed.isEmpty()) {
    return "Fine. Be that way!"
  }
  val regex = """[a-zA-Z]""".toRegex()
  if (trimmed.endsWith("?")) {
    if (regex.containsMatchIn(trimmed) && trimmed.toUpperCase() == trimmed) {
      return "Calm down, I know what I'm doing!"
    }
    return "Sure."
  }
  if (regex.containsMatchIn(trimmed) && trimmed.toUpperCase() == trimmed) {
    return "Whoa, chill out!"
  }
  return "Whatever."
}